File: _examples/typescript/classes.ts
class Person {
    private name: string;
    private age: number;
    
    constructor(name: string, age: number) {
        this.name = name;
        this.age = age;
    }
    
    greet(): void {
        console.log(`Hello, ${this.name}`);
    }
}

File: _examples/rust/classes.rs
struct Person {
    name: String,
    age: i32,
}

impl Person {
    fn new(name: String, age: i32) -> Self {
        Person { name, age }
    }
    
    fn greet(&self) {
        println!("Hello, {}", self.name);
    }
}